utils/diskusage/disk_usage.go

```diff
@@ -7,39 +7,32 @@ import "syscall"
 
 // DiskUsage contains usage data and provides user-friendly access methods.
 type DiskUsage struct {
-	stat *syscall.Statfs_t
+	// Free is the total free bytes on file system.
+	Free uint64
+	// Available is the total available bytes on file system to an unprivileged user.
+	Available uint64
+	// Size is the total size of the file system.
+	Size uint64
+	// Used is the bytes used in file system.
+	Used uint64
+	// Usage returns percentage of use on the file system.
+	Usage float64
 }
 
-// NewDiskUsage returns an object holding the disk usage of volumePath
-// or nil in case of error (invalid path, etc).
-func NewDiskUsage(volumePath string) *DiskUsage {
+// Statfs returns file system statistics.
+func Statfs(volumePath string) (DiskUsage, error) {
```

Comment:
@vijayvuyyuru do you know why previously `Statfs` returning an error was ignored?

Reply:
This is fine to change. I don't have a good answer for you, I think it was a mistake when I was implementing.

Reply:
One more question @vijayvuyyuru 

It looks like only these 2 values are ever used from this struct (available and size bytes) do you know if it is ok to remove the data we DON'T currently use?

Was there ever a requirement to add these other fields?

Reply:
There was not. This code was mostly copied from a small open source library so that we weren't reliant on an external depency for relatively straightforward logic. Feel free to remove those fields.